com.unity.uiwidgets/Runtime/engine/UIWidgetsPanelWrapper.cs

```diff
@@ -74,12 +74,107 @@ public partial class UIWidgetsPanelWrapper {
     static extern IntPtr UIWidgetsPanel_onRenderTexture(
         IntPtr ptr, int width, int height, float dpi);
 }
-
 #endif
-
     #endregion
+    
+    #region Platform: MacOs/iOS Specific Functionalities
+#if UNITY_EDITOR_OSX || UNITY_STANDALONE_OSX || UNITY_IOS
+    public partial class UIWidgetsPanelWrapper {
+        Texture _renderTexture;
+
+        public Texture renderTexture {
+            get { return _renderTexture; }
+        }
+        
+        //since both Metal and OpenGLCore are supported on Mac/iOS, we use this flag to separate the code paths for
+        //each backend respectively
+        bool isMetalBackend => SystemInfo.graphicsDeviceType == GraphicsDeviceType.Metal;
+
+        void _createRenderTexture(int width, int height, float devicePixelRatio) {
+            D.assert(_renderTexture == null);
+
+            if (!isMetalBackend) {
```

Comment:
Can we seperate metal and opengl 's logic into two scripts?  It looks a little confusing.

Reply:
The method `_createRenderTexture` is the unified interface we defined to create a render texture regardless of which backend (metal or opengl) is used. Moreover, we cannot use the `#UNITY_XXX` tricks to split the codes for different graphics backend since there is no such builtin directives in Unity. As the result, we have to put the codes for both metal and opengl inside this method.

However, maybe we can remedy this via ?
```
void _createRenderTexture(int width, int height, float devicePixelRatio) {
            D.assert(_renderTexture == null);
       if (!isMetalBackend) _createRenderTextureForOpenGL(width, height, devicePixelRatio);
       else _createRenderTextureForMetal(width, height, devicePixelRatio);
}
```

Reply:
If we can't seperate, just keep it.